wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +487,147 @@ def get_device_unique_id(device):
 
     def disconnect(self):
         self.logger.info("LenovoSystem disconnected")
+
+    def __getattr__(self, attr):
+        """
+        Tries to delegate a not found method to objects inside self._delegate
+        :raises AttributeError if ``attr`` is not found in any objects this should delegate
+        or if ``attr`` is found in 2 or more delegated classes/objects.
+        """
+        matches = []
+        for delegate_to in self._delegate:
+            if hasattr(delegate_to, attr):
+                matches.append(delegate_to)
+        if len(matches) == 1:
+            return getattr(matches[0], attr)
+        elif len(matches) > 1:
+            raise AttributeError("Ambiguous reference to attribute: {}".format(attr))
+        else:
+            raise AttributeError(attr)
+
+
+class LenovoSwitch(Entity):
+    """ Encapsulates all Lenovo Switches behavior """
+    API_PATH = "switches"
+
+    def __init__(self, switch_data, **kwargs):
+        super(LenovoSwitch, self).__init__(raw=switch_data, **kwargs)
+
+    @property
+    def product_name(self):
+        return self.raw.get("productName", None)
+
+    @property
+    def manufacturer(self):
+        return self.raw.get("manufacturer", None)
+
+    @property
+    def serial_number(self):
+        return self.raw.get("serialNumber", None)
+
+    @property
+    def part_number(self):
+        return self.raw.get("partNumber", None)
+
+    @property
+    def description(self):
+        return self.raw.get("description", None)
+
+    @property
+    def firmwares(self):
+        return self.raw.get('firmware', None)
+
+    @property
+    def power_status(self):
+        return self.raw.get('powerState', None).lower()
+
+    @property
+    def type(self):
+        return self.raw.get('type', None)
+
+    @property
+    def health_state(self):
+        return self.raw.get('overallHealthState', None)
+
+    @property
+    def ipv4_addresses(self):
+        return self.raw.get('ipv6Addresses', None)
+
+    @property
+    def ipv6_addresses(self):
+        return self.raw.get('ipv6Addresses', None)
+
+    @property
+    def ports(self):
+        return LenovoSwitch.get_switch_ports(self.raw)
+
+    @property
+    def ipv4_assignments(self):
+        return LenovoSwitch.get_switch_ipv6_assignments(self.raw)
+
+    @property
+    def ipv6_assignments(self):
+        return LenovoSwitch.get_switch_ipv6_assignments(self.raw)
+
+    @property
+    def _identifying_attrs(self):
+        return {"name": self.name}
```

Comment:
I'm thinking name and UUID should both be correlated here for identifying the same switch object.